include/avk/commands.hpp

```diff
@@ -1113,23 +1113,34 @@ namespace avk
 	{
 		avk::resource_argument<avk::semaphore_t> mWaitSemaphore;
 		avk::stage::pipeline_stage_flags mDstStage;
+		uint64_t mValue;
+
+		semaphore_wait_info& at_value(uint64_t aValue) {
+			mValue = aValue;
+			return *this;
+		}

```

Comment:
Hmm, I don't like this approach too much, because it leads to the impression that `stage >> semaphore` is a unit, whereas `semaphore + value` should be a unit.

Reply:
My most recent commit addresses this issue and `semaphore_wait_info` / `semaphore_signal_info` can now be created via
```c++
semaphore_signal_info foo = pipelineStageFlags >> (signaledSemaphore, newValue);
semaphore_wait_info bar = (waitSemaphore, waitValue) >> pipelineStageFlags;
```
However, to get this to work I had to overload `operator,(avk::resource_argument<avk::semaphore_t>, uint64_t)`. While I currently don't see any unwanted side effects this might have, it doesn't strike me as clean coding practice.

Reply:
Uff, `operator,` sounds very adventurous. And what's more, it even requires the parens around it, right? So it actually isn't really of any benefit over constructing `semaphore_value_info` directly via `semaphore_value_info{signaledSemaphore, newValue}` or `semaphore_value_info{waitSemaphore, waitValue}`. 

One more thing comes to mind which might be cooler: Overloading `opertator=` and `operator==`. What do you think? 
- `semaphore_signal_info foo = pipelineStageFlags >> signaledSemaphore = newValue;`
- semaphore_wait_info bar = waitSemaphore == waitValue >> pipelineStageFlags;` 

Is that expressive and well understandable?

Reply:
When you're investigating `operator=` and `operator==`, please make sure that you handle both cases:
-  `pipelineStageFlags >> signaledSemaphore = newValue` and `pipelineStageFlags >> (signaledSemaphore = newValue)`
- `waitSemaphore == waitValue >> pipelineStageFlags` and `(waitSemaphore == waitValue) >> pipelineStageFlags`

You might have to create different overloads for  `operator=` and `operator==` to handle both cases. But this is fine.

Reply:
Finally got to it and implemented the suggested change.

- `semaphore_signal_info foo = pipelineStageFlags >> signaledSemaphore = newValue;`
- `semaphore_wait_info bar = waitSemaphore >= waitValue >> pipelineStageFlags;`
- `semaphore_value_info baz = semaphore = value;`
- `semaphore_value_info quz = semaphore >= value;`

Are all valid expressions now. Note that I used `operator>=` instead of `operator==` since this is the technically more correct way to specify it.

I again had to make some less desireable changes to get this to compile. I will highlight those in seperate threads so we can discuss potential workarounds.